test/kongintegration/containers/httpbin.go

```diff
@@ -0,0 +1,53 @@
+package containers
+
+import (
+	"context"
+	"strconv"
+	"testing"
+
+	"github.com/docker/go-connections/nat"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/test"
+)
+
+// HTTPBin represents a docker container running the `kong/httpbin`.
+type HTTPBin struct {
+	container testcontainers.Container
+}
+
+// NewHTTPBin spawns a docker container running the `kong/httpbin` which can be used for proxy routing testing.
+func NewHTTPBin(ctx context.Context, t *testing.T) HTTPBin {
```

Comment:
nit: I rather prefer moving it above `type HTTPBin` to have method `IP(..)` closer to struct

Reply:
🤔 What would be the benefit of this? I think that we rather always stick to the following order:

1. struct definition
2. its constructor(s)
3. other methods

Reply:
> 🤔 What would be the benefit of this? I think that we rather always stick to the following order:
> 
> 1. struct definition
> 2. its constructor(s)
> 3. other methods

Agree, I believe this is the best we can do with Go.

Reply:
Yeah, I looked at the codebase and this is the most popular order in KIC. It's just a preference, even stdlib is not consistent about order. It's not something that I would argue about, let's do it as everywhere